haystack/utils/openai_utils.py

```diff
@@ -74,14 +74,24 @@ def _openai_text_completion_tokenization_details(model_name: str):
 
     :param model_name: Name of the OpenAI model.
     """
-    tokenizer_name = "gpt2"
-    if "davinci" in model_name:
-        max_tokens_limit = 4000
-        if USE_TIKTOKEN:
-            tokenizer_name = MODEL_TO_ENCODING.get(model_name, "p50k_base")
+    # Adding the new tokenizer for ChatGPT model and adjusting the `max_tokens_limit`
+    ## Ref: https://github.com/openai/tiktoken/blob/d0cc25c8a670941e38062cd67eef97c7f6bb74d3/tiktoken/model.py#L14
+
+    default_tokenizer_name = "gpt2"
+    max_tokens_limit = 2048
+    model_tokenizer = MODEL_TO_ENCODING.get(model_name)
```

Comment:
also as a heads up this change will most likely fail when tiktoken is not installed since `MODEL_TO_ENCODING` can only be imported if tiktoken is present. So you could handle this potentially by doing this 
```python
if USE_TIKTOKEN:
    model_tokenizer = MODEL_TO_ENCODING.get(model_name)
else:
    model_tokenizer = None
```

Reply:
Or we can put this line inside the first if condition, right!

Reply:
You can! However, you would no longer be able to check `if USE_TIKTOKEN and model_tokenizer:` since `model_tokenizer` would no longer be defined before the first if condition.

Reply:
Hello @sjrl sorry for annoying you, I am facing a problem regards the pushing; Would you mind checking this screenshot, please?

![image](https://user-images.githubusercontent.com/72295771/223823058-68b09b11-1bfe-4cf8-ae5a-77cb683fde38.png)

Reply:
I also found that this is one of the problems that prevent the came after everything is installed...

![image](https://user-images.githubusercontent.com/72295771/223889595-f06f0c2f-f175-4b24-be9e-70d2dfbb6126.png)

Reply:
Hmm @AI-Ahmed for the first one did this occur after you ran a merge or rebase of main? If so we can circumvent the github hooks for this push by adding `--no-verify` to your git push (e.g. `git push --no-verify -f ...`). Keep in mind this won't circumvent the checks we run on this PR in GitHub so this should be fine to do.

Reply:
@AI-Ahmed I'm not familiar with this installation error message. @ZanSara have you happened to see installation errors with `mmh3` on linux before?

Reply:
I will try this until you tell me if there is a solution for that.

Reply:
I had to reset everything I had done after I saw that there are problems with the files as I can see here